### PR TITLE
Add open graph meta tags for blog posts and documentation

### LINKED ISSUE #104: Add social meta data for blog posts
Right now our blog posts don't use any meta data for the blog posts, so we won't have unique social cards for blogs

image

We should already have the [frontmatter available](https://github.com/remix-austin/remixaustin-com/blob/main/app/routes/resource/get-blog-post.%24slug.ts#L17-L19) which we can pass into the [meta function](https://remix.run/docs/en/1.16.0/route/meta#page-context-in-meta-function).

We will also want to figure out a good way to select an image to use for things like the [og:image](https://ogp.me/#structured)

### PR DESCRIPTION
**Pull request by matt-hernandez**
Closes #104 

This PR adds Open Graph meta tags to blog posts. New front matter requirements have been added to blog posts in order to render the right meta tags. Documentation and code for the blogs have been updated as a result. All changes must be shipped together

### CODE REVIEW

**Comment by matt-hernandez** [comment]
@brookslybrand With a blog banner image of 1200 x 630, which is the most common size I found recommended online, here are the results from testing the staging branch on [this site](https://www.opengraph.xyz/). You can see that Facebook and LinkedIn are getting a little cut off on the sides, but Twitter and Discord look good.

Screenshot 2023-05-02 at 22 15 17

Screenshot 2023-05-02 at 22 15 22

Screenshot 2023-05-02 at 22 15 28

Screenshot 2023-05-02 at 22 15 34

However, with an image size of 1200 x 690, we get even better results. I'd argue that this is the best all around size. It strikes the best balance between looking good on most major sites while also not requiring us to make multiple image banners for every site. I'm also adding a note in the blog README that blog banner images should try to keep the focus of the image towards the center and save some breathing room around the sides.

Screenshot 2023-05-02 at 22 29 56

Screenshot 2023-05-02 at 22 30 04

Screenshot 2023-05-02 at 22 30 35

Screenshot 2023-05-02 at 22 30 45

File: app/routes/blog.$slug.tsx [hunk 1]
```diff
@@ -9,16 +10,43 @@ import type { Mdx } from "blog/parser";
...
+    "og:type": "article",
+    "og:author": author,
+    "og:published_time": new Date(date).toISOString(),
+    "og:title": title,
+    "og:description": description,
+    "og:image": imageUrl[0] === "/" ? `${origin}${imageUrl}` : imageUrl,
+    "og:image:type": "image/jpeg",
+    "og:image:width": "1200",
+    "og:image:height": "630",
+  };
```
Comment (brookslybrand) [inline app/routes/blog.$slug.tsx:44]: Let's add the `title` and `description` tags for good measure 🙂 

```suggestion
  return {
    title,
    description,
    "og:url": data.url,
    "og:type": "article",
    "og:author": author,
    "og:published_time": new Date(date).toISOString(),
    "og:title": title,
    "og:description": description,
    "og:image": imageUrl[0] === "/" ? `${origin}${imageUrl}` : imageUrl,
    "og:image:type": "image/jpeg",
    "og:image:width": "1200",
    "og:image:height": "630",
  };
```

File: app/routes/blog.$slug.tsx [hunk 2]
```diff
@@ -9,16 +10,43 @@ import type { Mdx } from "blog/parser";
...
+    post,
+    url: request.url,
+  };
+};
+
+/**
+ * TODO: When we update to Remix v2, we need to change this function according to this
+ * page in the docs: https://remix.run/docs/en/1.16.0/pages/v2#route-meta
+ * Blog tags can't be added to meta in Remix v1. We need to wait until v2.
+ */
```
Comment (brookslybrand) [inline app/routes/blog.$slug.tsx:27]: Hoping to make these updates soon, thanks for calling this out!

**Review by brookslybrand — APPROVED** [review]
Looks great! Thanks @matt-hernandez, and thanks for doing a bit of research on the best image size, I think those socials look sweet 😎 

Added just one thing that needs to be updated, otherwise I think this is ready to go and for us to start sharing!

**Pull request merged**